code/controller/WidgetController.php

```diff
@@ -109,6 +109,7 @@ public function editablesegment()
         }
         if (class_exists($className) && is_subclass_of($className, 'Widget')) {
             $obj = new $className();
+			$obj->write();// generate new id
```

Comment:
Can you move the comment above the line? The indentation on this also needs to be fixed up to match the rest of the code style